docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,474 @@
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  match:
+  - type: InternalVIP # Kuma will generate a domain
+    value: example.ext.svc.local
+    port: 80
+    protocol: http
+  - type: Domain # Existing domain
+    value: httpbin.com
+    port: 80
+    protocol: http
+  - type: CIDR
+    value: 10.1.1.0/24
+    port: 80
+    protocol: http
+  - type: IP
+    value: 192.168.0.1
+    port: 80
+    protocol: http
+  destination:
+    type: Regular # Regular|Passthrough|Extension
+    extension:
+      type: Lambda 
+      config: # type JSON
+        arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+    endpoints:
+    - address: 1.1.1.1
+      port: 12345
+    - address: example.com
+    - address: unix://....
+    tls:
+      enabled: true
+      version:
+        min: TLSv1_2 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+        max: TLSv1_3 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+      allowRenegotiation: false
+      verification:
+        skip: true # if this is true then subjectAltNames don't take effect
+        subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+          - example.com
+          - "spiffe://example.local/ns/local"
+      caCert: 
+        inline: 123
+      clientCert:
+        secret: 123
+      clientKey:
+        secret: 123
+status: 
+  vips:
+  - ip: 242.0.0.1
+    type: Kuma
+    hostname: example.ext.svc.local
+```
+* **spec**:
+  * **match**: defines a list of internalVIPs/domains/CIDR/IPs that should be routed through the sidecar
+    * **type**: type of the match, one of `InternalVIP`, `Domain`, `CIDR` and `IP` are available
+      * `InternalVIP`: allocates a VIP for a domain provided in the `value` field
+      * `Domain`: handles traffic to the specified domain
+      * `CIDR`: handles traffic to specified addresses range
+      * `IP` handles the traffic to specified IP
+    * **value**: depends of the type can be an existing domain, new domain name, CIDR or IP
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tls`: should be used when TLS traffic is originated by the client application
+      * `tcp`: WARNING: shouldn't be used when match has only domains. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **destination**: defines where matched requests should be routed
+    * **type**: defines what kind of destination is it, one of `Regular`, `Passthrough`, or `Extension`, (Default: `Passthrough`)
+      * `Regular`: allows creating a set of destination endpoints and `TLS` configuration, when defined section `extension` is not available.
```

Comment:
It would be nice to figure out a better name here. Maybe `Managed`? Because external service is managed by Envoy. I'm curious to hear from others

Reply:
I think I dislike `Managed` less